### DISCUSSION TITLE
Regex Dockerfile Versions Question

### DISCUSSION THREAD

**Discussion by xMikux** [renovatebot/renovate#18198] [Request Help] [answered]
Hi, I am trying to setup to update argument in dockerfile.
 ```dockerfile
# renovate: datasource=github-releases depName=cloudflare/cloudflared
ARG CLOUDFLARED_VERSION=2022.9.1
```
but when I use [regexManagers:dockerfileVersions](https://docs.renovatebot.com/presets-regexManagers/), it is not detected regex depend from Dependency Dashboard.

I change the regex match to 
``?<datasource>[a-z-]+?`` >``<depName>.*.``
```
# renovate: datasource=(?<datasource>[a-z-]+?) depName=(?<depName>.*.?)(?: (lookupName|packageName)=(?<packageName>[^\\s]+?))?(?: versioning=(?<versioning>[a-z-0-9]+?))?\\s(?:ENV|ARG) .+?_VERSION=\"?(?<currentValue>.+?)\"?\\s
```
it works, find cloudflared depend, but there have an error ``WARN: Error updating branch: update failure`` so still can't update.

I am not sure which step I do wrong, and sorry for my strange english.

**Comment by RahulGautamSingh** [comment, marked as answer]
Hey @xMikux ,
I tried imitating you issue and all seemed to work fine and a PR was created. Here is the [example repo](https://github.com/RahulGautamSingh-testing/d-18198). Do tell me if I missed something in the config. :)

Branch update failure can have other reason than regex mismatch so I recommend you to attach the logs for your run so that we can be sure it is regex issue.

**Reply by xMikux** [reply]
Thanks for the test!
Here is my [log.txt](https://github.com/renovatebot/renovate/files/9739450/log.txt) and [renovate](https://github.com/Docker-Collection/docker-cloudflared/blob/main/.github/renovate.json) config.
Debug level say ``DEBUG: depName mismatch(packageFile="Dockerfile", branch="renovate/cloudflare-cloudflared-2022.x")``

**Reply by xMikux** [reply]
After I recreate the repo, and reput all files, it back to normal, still don't know what caused it, but yep, it was fixed.
Thank you!
